.github/workflows/semantic-pr-check.yaml

```diff
@@ -0,0 +1,38 @@
+---
+name: "Semantic PR Title Check"
+# Ensure that the PR title conforms to the Conventional Commits and our choice of types and scopes, so that library version bumps can be detected automatically
+
+on:
+  pull_request_target:
+    types: [opened, edited, synchronize]
+
+permissions:
+  pull-requests: read
+
+jobs:
+  main:
+    name: Validate PR title
+    runs-on: ubuntu-latest
+    steps:
+      - uses: amannn/action-semantic-pull-request@v5
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          types: |
+            chore
+            ci
+            docs
+            feat
+            fix
+            perf
+            refactor
```

Comment:
Per https://github.com/canonical/operator/pull/1252 I think we want to add `revert` and `test`.

Reply:
Thanks, I've missed that.